### ISSUE TITLE
Mapping concept IDs to other vocabularies

### ISSUE THREAD

**Issue by cmungall** [CaseyTa/cohd#1]
Not sure if this is the right place to report...

I'm using http://smart-api.info/ui/?url=/api/metadata/6c33ed14d628a982c79fa36a75dbbbcf, very clear API thanks

I'm trying to find additional info on each concept in particular the diseases, it looks like the IDs assigned are local to this particular dataset. E.g. http://cohd.nsides.io/api/v1/omop/concepts?q=192855

```
{
  "results": [
    {
      "concept_id": 192855,
      "concept_name": "Cancer in situ of urinary bladder",
      "domain_id": "Condition"
    }
  ]
}
```

The ID 192855 doesn't seem to correspond to any vocabulary I know. My guess is that this came from snomed as AFAIK it is the only vocabulary that uses "cancer in situ" as a synonym for "carcinoma in situ".  The snomed ID for this is SCTID:92546004 which doesn't seem to be linked (or NCIT:C3644 or  UMLS:C0154091 or  ICD10:D09.0 ...).

I would be interested in using this API at the translator meeting next week. If no such mapping exists I can create a reasonably accurate one quickly (if you can provide a dump of all concepts that would be easier than me iterating through the API). Or if the mapping exists, would it be possible to expose this (just a file is fine - I can help expose via either this API, or via a separate API that uses our [MONDO](obofoundry.org/ontology/mondo.html) resource).

**Comment by cmungall** [comment]
I am iterating through the API just now hope that doesn't cause any overload issues. For every MONDO class I'm doing a query first on the primary name then syns, and moving onto the next class when I get a result. I pause for 0.5s after every hit.

**Comment by cmungall** [comment]
OK, preliminary mapping here:

https://raw.githubusercontent.com/monarch-initiative/mondo-build/master/src/ontology/reports/release/semantic-xref-pairs.tsv

**Comment by CaseyTa** [comment]
Hi Chris, sorry, I completely missed this issue. I've actually been talking to David Koslicki over the last couple weeks, and he suggested using OxO (https://www.ebi.ac.uk/spot/oxo/) to map to other ontologies. I just got those updates up and running. I checked about 10 mappings to / from Mondo, and the mappings from OxO agreed with yours minus 1. 

MONDO:0000022	enuresis, nocturnal	COHD	COHD:193874	equivalent

From Mondo to OMOP was the same mapping, but from OMOP to MONDO yielded MONDO:0010910 ("ENUR1"). I haven't done a more thorough check, but the mappings seem to be in reasonably good agreement. 

Also, our concept IDs come from the OMOP Common Data Model. I would suggest a prefix of OMOP as opposed to COHD in your mappings.

**Comment by cmungall** [comment]
Hi @CaseyTa - yep they agree as OxO gets the COHD mappings from MONDO, where I put them after making this ticket!

Not sure about the discrepancy, I suspect this is distance>1 when other non-exact mappings are introduced.

Is there a resolver for OMOP IDs?

**Comment by CaseyTa** [comment]
Hi @cmungall, actually I was completely unaware that OxO has COHD mappings. I don't see it (COHD or OMOP) in the UI as an available target and I did a quick test of the API call, but that returned no results. What I did was I mapped from OMOP standard concept IDs to concept IDs in ICD9CM, ICD10CM, SNOMEDCT, and MeSH (these mappings are already available in the OMOP CDM), then used OxO to map from these intermediates to the target ontology. 

I did't pay attention to the actual distance when I checked the mappings, but I most found results with distance <= 2. I changed one to distance 3 and it found an exact match with your mapping. 

The OMOP vocabulary can be downloaded from athena.ohdsi.org. It's a much larger set of concepts that those found in the original COHD release.
